extensions/wikia/LinkSuggest/LinkSuggest.class.php

```diff
@@ -172,19 +173,42 @@ static function getLinkSuggest(WebRequest $request) {
 			 * It uses fact that page titles can't start with lowercase letter.
 			 */
 			$pageTitlePrefilter = "";
-			if( strlen($queryLower) >= 2 ) {
-				$pageTitlePrefilter = "(
-							( convert(binary convert(page_title using latin1) using utf8) " . $db->buildLike(strtoupper($queryLower[0]) . strtolower($queryLower[1]) , $db->anyString() ) . " ) OR
-							( convert(binary convert(page_title using latin1) using utf8) " . $db->buildLike(strtoupper($queryLower[0]) . strtoupper($queryLower[1]) , $db->anyString() ) . " ) ) AND ";
-			} else if( strlen($queryLower) >= 1 ) {
-				$pageTitlePrefilter = "( page_title " . $db->buildLike(strtoupper($queryLower[0]) , $db->anyString() ) . " ) AND ";
+
+			// use mb_substring to get the first & second chars
+			// in case of multi-byte
+			$firstChar = mb_substr($queryLower, 0, 1);
+			$secondChar = mb_substr($queryLower, 1, 1);
+
+			if( mb_strlen( $queryLower ) >= 2 ) {
+				if ( LinkSuggest::containsMultibyteCharacters( $firstChar . $secondChar ) ) {
+					$pageTitlePrefilter = "(
+						( convert(binary convert(page_title using latin1) using utf8) LIKE convert(binary convert('" . strtoupper( $firstChar ) . strtolower( $secondChar ) . "%' using latin1) using utf8)) ) OR
+						( convert(binary convert(page_title using latin1) using utf8) LIKE convert(binary convert('" . strtoupper( $firstChar ) . strtoupper( $secondChar )  . "%' using latin1) using utf8) ) AND ";
+				} else {
+					$pageTitlePrefilter = "(
+						( convert(binary convert(page_title using latin1) using utf8) " . $db->buildLike( strtoupper( $firstChar ) . strtolower( $secondChar ) , $db->anyString() ) . " ) OR
+						( convert(binary convert(page_title using latin1) using utf8) " . $db->buildLike( strtoupper( $firstChar ) . strtoupper( $secondChar ) , $db->anyString() ) . " ) ) AND ";
+				}
+			} else if( mb_strlen($queryLower) >= 1 ) {
+				if ( LinkSuggest::containsMultibyteCharacters( $firstChar ) ) {
+					$pageTitlePrefilter = "( convert(binary convert(page_title using latin1) using utf8) LIKE convert(binary convert('" . $firstChar . "'% ) AND ";
+				} else {
+					$pageTitlePrefilter = "( convert(binary convert(page_title using latin1) using utf8) " . $db->buildLike(strtoupper( $firstChar ) , $db->anyString() ) . " ) AND ";
+				}
 			}
+
+			if ( LinkSuggest::containsMultibyteCharacters( $queryLower ) ) {
+				$pageTitleLikeClause = "convert(binary convert('{$queryLower}%' using latin1) using utf8)";
+			} else {
+				$pageTitleLikeClause = "'{$queryLower}%'";
+			}
+
 			// TODO: use $db->select helper method
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this "TODO" comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1135)